File: 프로그래머스/unrated/181854. 배열의 길이에 따라 다른 연산하기/배열의 길이에 따라 다른 연산하기.java
class Solution {
    public int[] solution(int[] arr, int n) {
        int[] answer = new int[arr.length];
        // int의 길이는 변함 없으므로 arr과 동일한 길이로 선언한다
        for(int i = 0; i < arr.length; i++) {
            if(arr.length % 2 != 0) {
            // arr의 길이가 홀수 일때,
                if(i % 2 == 0) {
                    answer[i] += arr[i] + n;
                }   // 짝수번째 원소에 n을 더하고,
                else {
                    answer[i] += arr[i];
                }   // 홀수번째는 기존의 원소를 넣는다
            }
            if(arr.length % 2 == 0) {
            // arr의 길이가 짝수라면,
                if(i % 2 != 0) {
                    answer[i] += arr[i] + n;
                }   // 홀수번째 원소에 n을 더하고,
                else {
                    answer[i] += arr[i];
                }   // 짝수번째 원소는 그대로 넣는다
            }
        }
        return answer;
    }
}

File: 프로그래머스/0/181854. 배열의 길이에 따라 다른 연산하기/배열의 길이에 따라 다른 연산하기.js
function solution(arr, n) {
    var answer = new Array(arr.length);
    // arr과 동일한 길이로 배열을 선언합니다.
    
    for (let i = 0; i < arr.length; i++) {
        if (arr.length % 2 !== 0) {
            // arr의 길이가 홀수일 때,
            if (i % 2 === 0) {
                answer[i] = arr[i] + n;
            } else {
                answer[i] = arr[i];
            } // 짝수 번째 원소에 n을 더하고, 홀수 번째는 기존의 원소를 넣습니다.
        } else {
            // arr의 길이가 짝수라면,
            if (i % 2 !== 0) {
                answer[i] = arr[i] + n;
            } else {
                answer[i] = arr[i];
            } // 홀수 번째 원소에 n을 더하고, 짝수 번째 원소는 그대로 넣습니다.
        }
    }
    return answer;
}